### PR TITLE
feat(builder): allow script to build multiple resources

### PR DESCRIPTION
**Pull request by CENSOR37**
I would greatly appreciate any suggestions or improvements you might have =D

### CODE REVIEW

File: scripts/compiler.js [hunk 1]
```diff
@@ -19,21 +19,26 @@ const SWC_CONFIG = {
     sourceMaps: false,
 };
 
-const startTime = Date.now();
-const filesToCompile = glob.sync('./src/core/**/*.ts');
+async function BuildTargetResource(name) {
```
Comment (Stuyk) [inline scripts/compiler.js]: `BuildTargetResource` should be `buildTargetResource`

It is not a class and not invoked with `new` so common convention is `camelCase`

File: scripts/compiler.js [hunk 2]
```diff
@@ -19,21 +19,26 @@ const SWC_CONFIG = {
...
-    const finalPath = filePath.replace('src/', 'resources/').replace('.ts', '.js');
-    const compiled = swc.transformFileSync(filePath, SWC_CONFIG);
-    fs.outputFileSync(finalPath, compiled.code, { encoding: 'utf-8' });
-    compileCount += 1;
+    console.log(`[${name}] Has built ${compileCount} files in ${Date.now() - startTime}ms`)
 }
 
-console.log(`${compileCount} Files Built | ${Date.now() - startTime}ms`);;
+for (let resource of resources) {
+    BuildTargetResource(resource);
```
Comment (Stuyk) [inline scripts/compiler.js]: Update name

File: scripts/copy.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
 import fs from 'fs-extra'
 import * as glob from 'glob';
-import { normalizeFilePath } from './shared.js';
+import { resources, normalizeFilePath } from './shared.js';
 
-const startTime = Date.now();
-const files = glob.sync(['src/core/**/*.!(ts)'])
+async function CopyResourceAssets(name) {
```
Comment (Stuyk) [inline scripts/copy.js]: camelCase for function name

File: scripts/copy.js [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
+        filesCopied += 1;
+    }
+
+    console.log(`[${name}] | ${filesCopied} Files Moved | ${Date.now() - startTime}ms`);
 }
 
-console.log(`${filesCopied} Files Moved | ${Date.now() - startTime}ms`);
\ No newline at end of file
+for (let resource of resources) {
+    CopyResourceAssets(resource);
```
Comment (Stuyk) [inline scripts/copy.js]: camelCase for function name

File: scripts/shared.js [hunk 5]
```diff
@@ -12,4 +12,8 @@ export async function sleep(ms) {
     return new Promise((resolve) => {
         setTimeout(resolve, ms);
     })
-}
\ No newline at end of file
+}
+
+export const resources = [
```
Comment (Stuyk) [inline scripts/shared.js]: Preferably we shouldn't be specifying what resources to compile.

Instead I suggest we take a look at the `.toml` file for the server and extract resources that way.

If we need to ignore certain folders from compiling maybe consider adding a file inside of a resource folder called `.nocompile` or `.ignore` or something of the sort and immediately skip that resource entirely if it is present.

File: scripts/transform.js [hunk 6]
```diff
@@ -20,63 +18,72 @@ const funcsToIgnore = [
     '=>',
 ];
 
-for (let filePath of filePaths) {
-    const fileContents = fs.readFileSync(filePath, { encoding: 'utf-8' });
-    const splitContents = fileContents.split(/\r?\n/);
+async function TransformResource(name) {
```
Comment (Stuyk) [inline scripts/transform.js]: `camelCase`

File: scripts/transform.js [hunk 7]
```diff
@@ -20,63 +18,72 @@ const funcsToIgnore = [
...
-        continue;
+        const finalFile = splitContents.join('\r\n');
+        fs.writeFileSync(filePath, finalFile, { encoding: 'utf-8' });
     }
+}
 
-    const finalFile = splitContents.join('\r\n');
-    fs.writeFileSync(filePath, finalFile, { encoding: 'utf-8' });
+for (let resource of resources) {
+    TransformResource(resource);
```
Comment (Stuyk) [inline scripts/transform.js]: `camelCase`

**Review by Stuyk — CHANGES_REQUESTED** [review]

**Comment by CENSOR37** [comment]
Hey, thanks for your advice!, I've made the changes based on all your suggestions. =D

[refactor(builder): using camelCase for function name](https://github.com/Stuyk/altv-typescript/pull/14/commits/6b35d4f2c10973101250b8114d639f4f0893da97) https://github.com/Stuyk/altv-typescript/pull/14/commits/6b35d4f2c10973101250b8114d639f4f0893da97


[refactor: reference target resource from server config](https://github.com/Stuyk/altv-typescript/pull/14/commits/30f91841168829b2056cc2c4eb51b58abb7a9c22) https://github.com/Stuyk/altv-typescript/pull/14/commits/30f91841168829b2056cc2c4eb51b58abb7a9c22

**Comment by Stuyk** [comment]
Looks much better. Only thing left is documenting it into the `README` file.

**Comment by CENSOR37** [comment]
> Looks much better. Only thing left is documenting it into the `README` file.

Updated the readme! Let me know what you think. Thanks!

**Review by Stuyk — APPROVED** [review]

**Pull request merged**